libs/resource_partitioner/examples/guided_pool_test.cpp

```diff
@@ -302,7 +276,40 @@ int main(int argc, char* argv[])
         }
     }
 
-    std::cout << "[main] resources added to thread_pools \n";
+    std::cout << "[rp_callback] resources added to thread_pools \n";
+}
+
+// the normal int main function that is called at startup and runs on an OS thread
+// the user must call hpx::init to start the hpx runtime which will execute hpx_main
+// on an hpx thread
+int main(int argc, char* argv[])
+{
+    hpx::program_options::options_description desc_cmdline("Test options");
+    desc_cmdline.add_options()("pool-threads,m",
+        hpx::program_options::value<int>()->default_value(1),
+        "Number of threads to assign to custom pool");
+
+    // HPX uses a boost program options variable map, but we need it before
+    // hpx-main, so we will create another one here and throw it away after use
+    hpx::program_options::variables_map vm;
+    hpx::program_options::store(
+        hpx::program_options::command_line_parser(argc, argv)
+            .allow_unregistered()
+            .options(desc_cmdline)
+            .run(),
+        vm);
+
+    pool_threads = vm["pool-threads"].as<int>();
+
+    // Setup the init parameters
+    hpx::init_params init_args;
+    init_args.argc = argc;
+    init_args.argv = argv;
+    init_args.desc_cmdline_ptr = std::make_shared<
+        hpx::program_options::options_description>(desc_cmdline);
```

Comment:
I really would like to investigate options that would avoid using a `shared_ptr` here. Also, we should use constructors on `init_params` as much as possible.

Reply:
Yeah, the `shared_ptr` is just a temporary solution. It's not intended to stay. As for the constructors, I actually prefer leaving them out. Which ones would you define? And should they always essentially have the same signatures as the corresponding `init/start` functions? Assigning them directly like this is IMO much easier to maintain.

Reply:
Well, I would supply argc/argv directly to `hpx::init`/`hpx::start` _only_ (i.e. not make them part of the `init_params`). Those are too common to go in there. I would prefer:
```
hpx::init_params params = {…};
hpx::init(argc, argv, params);
```
over
```
hpx::init_params params = {argc, argv, ...};
hpx::init(params);
```
any day. Probably the same for `hpx_main`. But I think we agreed on that above already.

For the other parameters, I think I could live with initializing them one by one or through compound initialization.

Reply:
Yes, I think we agree! The entrypoint and argc/argv should not be duplicated in `init_params` and in the `init` arguments. The PR doesn't 100% reflect the last discussions we've had here. Sorry for any potential confusion.